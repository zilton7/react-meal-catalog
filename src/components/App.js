import Nav from "./Nav";
import ItemPanel from "./ItemPanel";

const App = () => {
  return (
    <div className="App">
      <Nav />
      <ItemPanel />
    </div>
  );
};

export default App;
