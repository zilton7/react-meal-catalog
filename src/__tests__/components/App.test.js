import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import App from '../../components/App';
import store from '../../store/index';

afterEach(cleanup);

it("Takes website's snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <App />
      </Provider>,
    ),
  ).toMatchSnapshot();
});

it("Displays website's title", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
  const element = screen.getByText(/Veggie Meal Catalog/i);
  expect(element).toBeInTheDocument();
});

it('Displays search bar', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
  const element = screen.getByPlaceholderText(/Search for Meal/i);
  expect(element).toBeInTheDocument();
});
