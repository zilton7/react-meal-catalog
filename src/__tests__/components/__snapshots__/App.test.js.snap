// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes website's snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav>
      <h1>
        Veggie Meal Catalog
      </h1>
    </nav>
    <div>
      <div
        class="search-wrapper"
      >
        <input
          placeholder="Search for Meal"
          type="text"
          value=""
        />
      </div>
      <div
        class="meals"
      />
    </div>
  </div>
</DocumentFragment>
`;
